File: xkcd/src/main/java/xyz/jienan/xkcd/list/XkcdListContract.java
package xyz.jienan.xkcd.list;

import java.util.List;

import xyz.jienan.xkcd.XkcdPic;
import xyz.jienan.xkcd.base.BasePresenter;
import xyz.jienan.xkcd.base.BaseView;

public interface XkcdListContract {

    interface View extends BaseView<Presenter> {

        void updateData(List<XkcdPic> xkcdPics);

        void setLoading(boolean isLoading);

        void showScroller(int visibility);

        void isLoadingMore(boolean isLoadingMore);
    }

    interface Presenter extends BasePresenter {

        void loadList(int startIndex);

        boolean hasFav();

        void loadFavList();

        void loadPeopleChoiceList();

        boolean lastItemReached(long index);
    }
}

File: xkcd/src/main/java/xyz/jienan/xkcd/list/contract/XkcdListContract.kt
package xyz.jienan.xkcd.list.contract

import xyz.jienan.xkcd.base.BaseView
import xyz.jienan.xkcd.list.activity.BaseListView
import xyz.jienan.xkcd.list.presenter.ListPresenter
import xyz.jienan.xkcd.model.XkcdPic

interface XkcdListContract {

    interface View : BaseView<ListPresenter>, BaseListView {

        fun updateData(pics: List<XkcdPic>)

        fun isLoadingMore(isLoadingMore: Boolean)
    }
}
